### PR TITLE
RHEL8: Exclude testing with RHEL8 guest/host in some cases

### PR DESCRIPTION
**Pull request by sitoliu**
ID: 1625798
Sign-off-by: Sitong Liu <siliu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
@yanan-fu @xiywang Please help review, thanks.

**Comment by xiywang** [comment]
Could you specify the cases name you deprecated in RHEL8 in commit log?

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 1]
```diff
@@ -24,15 +24,25 @@
...
+            no RHEL.8
+            no Host_RHEL.m8
             pci_model = e1000
             match_string = "Gigabit Ethernet Controller"
             nics = ""
             extra_params += "-net none"
+        - nic_e1000e:
+            no ppc64 ppc64le
+            pci_model = e1000e
+            match_string = "Intel 82574L GbE Controller"
```
Comment (xiywang) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: Anywhere used this string or any other special purpose? I didn't catch it in nic_hotplug.py.
If it's redundant, please remove it.

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 2]
```diff
@@ -24,15 +24,25 @@
             extra_params += "-net none"
         - nic_e1000:
             no ppc64 ppc64le
+            no RHEL.8
+            no Host_RHEL.m8
```
Comment (yanan-fu) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: RHEL8 kernel don't have e1000 driver will only effect RHEL8 guest.
It is ok to boot a RHEL7 guest with e1000 device on RHEL8 host.
So, `no Host_RHEL.m8` is in-correct,  `no RHEL.8` is enough.

**Comment by yanan-fu** [comment]
id `1585096` is needed  for the second commit "multi_nic_hotplug" .
And i think these two commits are not for a related issue, split  them into two patch is better. it is up to.

**Comment by vivianQizhu** [comment]
@yanan-fu I suppose one patch is okay, since it is tiny fix and modifying same file. Thanks.

**Comment by yanan-fu** [comment]
@vivianQizhu That is not true, In fact, the two commits modify different file, one for `compliebench` and one for `multi_nic_hotplug`. 
Anyway, it is not a big problem.

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 3]
```diff
@@ -24,15 +24,23 @@
...
+            no RHEL.8
             pci_model = e1000
             match_string = "Gigabit Ethernet Controller"
             nics = ""
             extra_params += "-net none"
+        - nic_e1000e:
+            no ppc64 ppc64le
+            pci_model = e1000e
+            nics = ""
+            extra_params += "-net none"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: Not investigate, but I guess here need `match_string`

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 4]
```diff
@@ -24,15 +24,23 @@
             extra_params += "-net none"
         - nic_e1000:
             no ppc64 ppc64le
+            no RHEL.8
             pci_model = e1000
             match_string = "Gigabit Ethernet Controller"
```
Comment (xiywang) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: how about remove all the "match_string"?

File: qemu/tests/cfg/multi_nics_hotplug.cfg (hunk 3 above)
Comment (xiywang) [reply qemu/tests/cfg/multi_nics_hotplug.cfg]: it's not used now.
it should be deprecated code after some code changes previously.

**Comment by yanan-fu** [comment]
Agree with @xiywang , `match_string` is useless, and test pass without it.
(1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.nic_e1000e: PASS (1221.93 s)

**Comment by sitoliu** [comment]
(1/5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.nic_8139: PASS (1162.82 s)
 (2/5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.nic_virtio: PASS (1163.83 s)
 (3/5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.nic_e1000e: PASS (1174.15 s)
 (4/5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.combination: PASS (1155.79 s)
 (5/5) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.x86_64.io-github-autotest-qemu.multi_nics_hotplug.sameid: PASS (725.06 s)

**Comment by xiywang** [comment]
LGTM.

Acked-by: Xiyue Wang xiywang@redhat.com

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 5]
```diff
@@ -12,27 +12,32 @@
...
             nics = ""
             extra_params += "-net none"
         - nic_e1000:
             no ppc64 ppc64le
+            no RHEL.8
             pci_model = e1000
-            match_string = "Gigabit Ethernet Controller"
+            nics = ""
+            extra_params += "-net none"
+        - nic_e1000e:
```
Comment (luckyh) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: `e1000e` was introduced to qemu from `2.6.0`, so here should add param `required_qemu` to restrict the version to test, like: https://github.com/autotest/tp-qemu/blob/48ad741de8790be76fc162c92e3cd9bf0a525e8d/qemu/tests/cfg/qemu_img.cfg#L132

File: qemu/tests/cfg/multi_nics_hotplug.cfg [hunk 6]
```diff
@@ -12,27 +12,32 @@
...
+            pci_model = e1000e
             nics = ""
             extra_params += "-net none"
         - combination:
             no ppc64 ppc64le
             pci_model = virtio-net-pci
             pci_model_hotplug_nic1 = rtl8139
             pci_model_hotplug_nic3 = e1000
+            RHEL.8:
+                pci_model_hotplug_nic3 = e1000e
```
Comment (luckyh) [inline qemu/tests/cfg/multi_nics_hotplug.cfg]: The same at here.

File: qemu/tests/cfg/multi_nics_hotplug.cfg (hunk 5 above)
Comment (sitoliu) [reply qemu/tests/cfg/multi_nics_hotplug.cfg]: @luckyh Updated it, thanks!

**Pull request merged**